pyk/src/pyk/ktool/krun.py

```diff
@@ -205,9 +270,11 @@ def _krun(
     # ---
     check: bool = True,
     pipe_stderr: bool = True,
+    pipe_stdout: bool = True,
     logger: Logger | None = None,
     bug_report: BugReport | None = None,
     debugger: bool = False,
+    proof_hint: bool = False,
```

Comment:
Since there's `run_proof_hint_process`, for the sake of simplicity, it'd better not to expose this functionality in `_krun`.

Reply:
The idea to expose `--proof-hints` in PyK is to enable it for KWASM and KEVM. KWASM uses `_krun` see [here](https://github.com/runtimeverification/wasm-semantics/blob/c4fc79c684ca9007547a74d768a4e743e3f1431f/pykwasm/src/pykwasm/scripts/kwasm.py#L48) while KEVM uses `Krun.run` see [here](https://github.com/runtimeverification/evm-semantics/blob/25be7669a6d42341d563cf53354ba0bbd29d9372/kevm-pyk/src/kevm_pyk/__main__.py#L593).

Reply:
Therefore, I expect to see these parameters exposed so we can set them on the semantics caller.  Unless there's a simpler way to do it that I'm not aware of. In this case, I would be happy to adapt the code to use it!

Reply:
There's no need to extend behavior here, you'll need to add new parameters to these functions / add a new command in KWASM / KEVM (whatever the maintainers there prefer) anyways.

Reply:
Looking to the function calls on the links of previous message, can you point to me how krun would be aware that I want to output Hints?

If not by exposing this variable I can only think in passing the output type as HINTS.

Reply:
> can you point to me how krun would be aware that I want to output Hints

By calling `KRun.run_proof_hint`, and not `_krun` / `KRun.run` in the handler. The handlers get their arguments from a custom command line layer, so you'll need to add a new command or a new option for proof hints.